pkg/kademlia/routinggraph/routinggraph.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package routinggraph
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+)
+
+func extendPrefix(prefix string, bit bool) string {
+	if bit {
+		return prefix + "1"
+	}
+	return prefix + "0"
+}
+
+// Draw writes the routing graph obtained using a GetBucketListResponse in the specified file
+func Draw(file *os.File, info *pb.GetBucketListResponse) (err error) {
+	_, err = file.Write([]byte("digraph{\nnode [shape=plaintext, fontname=\"Courier\"];edge [dir=none];rankdir=LR;\n"))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_, errWrite := file.Write([]byte("}\n"))
+		err = errs.Combine(err, errWrite)
+		err = errs.Combine(err, file.Close())
+	}()
+	var buf bytes.Buffer
+	buckets := info.GetBuckets()
+	addBucketsToGraph(buckets, 0, &buf, "")
+	_, err = buf.WriteTo(os.Stdout)
+	return err
+}
+
+func splitBucket(buckets []*pb.GetBucketListResponse_Bucket, bitDepth int) ([]*pb.GetBucketListResponse_Bucket, []*pb.GetBucketListResponse_Bucket) {
```

Comment:
You can use named return here. `(left, right []*pb.GetBucketListResponse_Bucket)`. Also move downwards below printing funcs.